File: src/Backend/Backend/Models/Dto/RegisterDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Backend.Models.Dto
{
    public class RegisterDto
    {
        [Required]
        public string FirstName { get; set; }
        public string LastName { get; set; }

        [Required, EmailAddress]
        public string Email { get; set; }

        [Required, MinLength(8)]
        public string Password { get; set; }

        [Required, MinLength(8)]
        public string ConfirmPassword { get; set; }

        [Required]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public UserRole Role { get; set; }
    }
}

File: src/WebApp/src/app/Models/RegisterDto.ts
export class RegisterDto{

    constructor(
        private username : string,
        private email : string,
        private password : string,
        private confirmPassword : string,
        private role? : string
    ) {}

    getUsername() : string {
        return this.username;
    }

    getEmail() : string{
        return this.email;
    }

    getPassword() : string{
        return this.password;
    }

    getConfirmPassword() : string{
        return this.confirmPassword;
    }

    getRole() : string{
        return this.role;
    }

}